python/lsst/ts/observatory/control/maintel/mtcalsys.py

```diff
@@ -0,0 +1,567 @@
+        self.fiberspectrograph_blue_index = 1
+        self.fiberspectrograph_red_index = 2
+        self.linearstage_led_select_index = 1
+        self.linearstage_led_focus_index = 2
+        self.linearstage_laser_focus_index = 3
+        self.linearstage_select_index = 4
+
+        super().__init__(
+            components=[
+                "TunableLaser",
+                "LEDProjector",
+                f"FiberSpectrograph:{self.fiberspectrograph_blue_index}",
+                f"FiberSpectrograph:{self.fiberspectrograph_red_index}",
+                f"Electrometer:{self.electrometer_projector_index}",
+                f"LinearStage:{self.linearstage_led_select_index}",
+                f"LinearStage:{self.linearstage_led_focus_index}",
+                f"LinearStage:{self.linearstage_laser_focus_index}",
+                f"LinearStage:{self.linearstage_select_index}",
+            ],
+            domain=domain,
+            log=log,
+            intended_usage=intended_usage,
+        )
+
+        self.mtcamera = mtcamera
+        self.ls_select_led_location = 0.0  # um
+        self.ls_select_laser_location = 20.0  # um
+
+        self.laser_enclosure_temp = 20.0  # C
+
+        # ALl this needs to go in a config yaml file somewhere.
+        self.led_per_filter = {
+            "u": ["M385L3"],
+            "g": ["M455L4", "M505L4"],
+            "r": ["M565L3", "M660L4"],
+            "i": ["M730L5", "M780LP1"],
+            "z": ["M850L3", "M940L3"],
+            "y": ["M970L4"],
+        }
+        self.ls_led_locations = {
+            "u": 0.0,
+            "g": 1.0,
+            "r": 2.0,
+            "i": 3.0,
+            "z": 4.0,
+            "y": 5.0,
+        }
+        self.ls_led_focus = {"u": 0.0, "g": 1.0, "r": 2.0, "i": 3.0, "z": 4.0, "y": 5.0}
+
+        self.exptime_dict: dict[str, float] = dict(
+            camera=0.0,
+            electrometer=0.0,
+            fiberspectrograph=0.0,
+        )
+        self.delay = 1
+
+    def ls_laser_focus(self, wavelength: float) -> float:
+        """Function for evaluating the distance of the focus linear stage"""
+        return 10.0
+
+    async def change_wavelength(self, wavelength: float) -> None:
+        """Change the TunableLaser wavelength setting"""
+
+        task_wavelength = self.rem.tunablelaser.cmd_changeWavelength.set_start(
+            wavelength=wavelength, timeout=self.long_long_timeout
+        )
+        task_focus = self.linearstage_laser_focus.cmd_moveAbsolute.set_start(
+            distance=self.ls_laser_focus(wavelength)
+        )
+
+        await asyncio.gather(task_wavelength, task_focus)
+
+    async def is_ready_for_flats(self) -> bool:
+        """Add doctring"""
+        # TODO (DM-44310): Implement method to check that the
+        # system is ready for flats.
+        return True
+
+    async def setup_calsys(self, sequence_name: str) -> None:
+        """If monochromatic flats, check that laser can be enabled,
```

Comment:
This docstring is trying to explain what the function does but it doesn't say what it is supposed to do. Maybe start with:

> Setup the calibration system.

Also, please document the input parameters.